sklearn/model_selection/_search.py

```diff
@@ -563,6 +581,39 @@ def inverse_transform(self, Xt):
         self._check_is_fitted('inverse_transform')
         return self.best_estimator_.inverse_transform(Xt)
 
+    def _generate_warm_start_groups(self, candidate_params):
+        """Yield lists of parameter settings to perform warm start within
+
+        Groups by keys not specified in use_warm_start
+        """
+        use_warm_start = getattr(self, 'use_warm_start', None) or ()
+        if not use_warm_start:
+            for parameters in candidate_params:
+                yield [parameters]
+            return
+
+        # we hide use_warm_start parameters' values so that they share a group
+        # if all other parameters match
+        if isinstance(use_warm_start, str):
+            use_warm_start = {use_warm_start: None}
+        else:
+            use_warm_start = {k: None for k in use_warm_start}
+
+        prev_key = None
+        group = []
+        for parameters in candidate_params:
+            param_key = parameters.copy()
+            param_key.update(use_warm_start)
+            if param_key == prev_key:
```

Comment:
(Mostly a note for myself): that breaks if the parameter is e.g. a numpy array.

Pretty sure it would break too if it were an estimator